libc/string/string_errsig.c

```diff
@@ -175,3 +213,107 @@ TEST_GROUP_RUNNER(string_errsign)
 	RUN_TEST_CASE(string_errsign, strsignal_basic);
 	RUN_TEST_CASE(string_errsign, strsignal_real_time);
 }
+
+
+TEST_GROUP(string_perror);
+
+
+TEST_SETUP(string_perror)
+{
+	errno = 0;
+}
+
+
+TEST_TEAR_DOWN(string_perror)
+{
+}
+
+TEST(string_perror, perror_basic)
+{
+
+	errno = 0;
+	char msg[] = "Some error message";
+	char *res;
+	char exp[100];
+	sprintf(exp, "Some error message: %s\n", strerror(errno));
+
+	res = perrorToFile(msg);
+
+	TEST_ASSERT_EQUAL_STRING(exp, res);
+	free(res);
+}
+
+
+TEST(string_perror, perror_empty_message)
+{
+	char *res;
+
+	errno = 31;
+
+	char exp[50];
+	sprintf(exp, "%s\n", strerror(errno));
+	TEST_ASSERT_EQUAL_STRING(exp, res = perrorToFile(""));
+	free(res);
+}
+
+
+TEST(string_perror, perror_every_ascii)
+{
+	char *msg = testdata_createCharStr(257);
+
+
+	char *res;
+	errno = 8;
+	res = perrorToFile(msg);
+
+	char expected[356];
+	sprintf(expected, "%s: %s\n", msg, strerror(errno));
+	TEST_ASSERT_EQUAL_STRING(expected, res);
+	free(res);
+	free(msg);
+}
+
+
+TEST(string_perror, perror_huge_argument)
+{
+	char *msg = testdata_hugeStr;
```

Comment:
Remember to check for warnings and try to find out why they are occurring
![image](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/116297248/694cdb90-6045-4cf0-bcb5-e3cb3d066b1f)